src/carnot/core/data/auto_retrieval/_internal/query_optimization.py

```diff
@@ -0,0 +1,57 @@
+#Internal query optimization (physical plans + cost model)
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import Any, List
+
+from .index_management import IndexManagementPipeline
+from .query_planning import LogicalPlan
+
+
+@dataclass
+class PhysicalOperator:
+    """Executable operator with concrete index choices and parameters."""
+    impl_type: str
+    params: Any
+
+
+@dataclass
+class PhysicalPlan:
+    """Executable physical plan chosen from candidate implementations."""
+    operators: List[PhysicalOperator]
```

Comment:
Shouldn't a plan be a tree?

Reply:
Similar to the comment I left in `query_planning.py` (in a response to Tianyu's comment), I'd recommend taking a look at PZ's implementation of these classes to help create a guide for your implementation here.